workers/unity/Assets/Fps/Scripts/Editor/GameLogic/MapQualitySettingsInspector.cs

```diff
@@ -0,0 +1,169 @@
+            foreach (var qualityName in QualitySettings.names)
+            {
+                var newTileQualitySetting = new MapQualityLevelData();
+                newTileQualitySetting.QualityName = qualityName;
+                newTileQualitySetting.CheckoutDistance = MapQualitySettings.DefaultCheckoutDistance;
+                foreach (var setting in tileSettings.Settings)
+                {
+                    if (setting.QualityName != qualityName)
+                    {
+                        continue;
+                    }
+
+                    newTileQualitySetting.CheckoutDistance = setting.CheckoutDistance;
+                    break;
+                }
+
+                newQualities.Add(newTileQualitySetting);
+            }
+
+            tileSettings.Settings = newQualities;
+        }
+    }
+
+    private void DrawTileQualityProperties()
+    {
+        var settings = target as MapQualitySettings;
+        var qualitySettingsList = serializedObject.FindProperty("Settings");
+        for (var i = 0; i < qualitySettingsList.arraySize; i++)
+        {
+            var qualitySettings = qualitySettingsList.GetArrayElementAtIndex(i);
+            var qualitySettingsName = qualitySettings.FindPropertyRelative("QualityName").stringValue;
+            var qualitySettingsCheckoutDistance = qualitySettings.FindPropertyRelative("CheckoutDistance");
+
+            var style = settings.ShouldApplyDrawDistance && i == QualitySettings.GetQualityLevel()
+                ? activeQualityStyle
+                : EditorStyles.numberField;
+
+            GUI.backgroundColor = settings.ShouldApplyDrawDistance && i == activeQualityLevel
+                ? new Color(0, 1f, 0)
+                : Color.white;
+
+
+            qualitySettingsCheckoutDistance.floatValue =
+                EditorGUILayout.FloatField(qualitySettingsName, qualitySettingsCheckoutDistance.floatValue, style);
+        }
+
+        GUI.backgroundColor = Color.white;
+    }
+
+    private void DrawQualityPreviewButtons()
+    {
+        GUILayout.BeginVertical();
+
+        var settings = target as MapQualitySettings;
+
+        for (var i = 0; i < QualitySettings.names.Length; i++)
+        {
+            GUI.backgroundColor = i == activeQualityLevel
+                ? Color.green
+                : Color.white;
+            if (!GUILayout.Button(QualitySettings.names[i]))
+            {
+                continue;
+            }
+
+            activeQualityLevel = i;
+
+            QualitySettings.SetQualityLevel(i);
+            settings.OnValidate();
+        }
+
+        GUI.backgroundColor = Color.white;
+        GUILayout.EndVertical();
+    }
+
+    private void OnEnable()
+    {
+        SetQualityStyle();
+    }
+
+    private void SetQualityStyle()
+    {
+        // If scripts are recompiled, Unity seems to have a delay before EditorStyles is safe to query
+        // So, keep trying until it's ready.
```

Comment:
😱

Reply:
I know right? Haha. We could just remove the use of styles altogether (they only highlight what the current quality mode is set to, and we can do that in another way).

Reply:
I'm going to remove the use of editor styles, and just use GUI.color to highlight instead